lib/active_merchant/billing/gateways/mit.rb

```diff
@@ -0,0 +1,244 @@
+        add_payment(post, payment)
+        # Address not required
+        add_address(post, payment, options)
+        add_customer_data(post, options)
+        post[:key_session] = self.options[:key_session]
+
+        post_to_json = post.to_json
+        puts(post_to_json)
+        post_to_json_encrypt = encrypt(post_to_json, self.options[:key_session])
+
+        final_post = post_to_json_encrypt + '-' + self.options[:user]
+        commit('sale', final_post)
+      end
+
+      def capture(money, authorization, options = {})
+        puts('---------------------------------------------')
+        puts('Capture')
+        puts('---------------------------------------------')
+        post = {
+          operation: 'Capture',
+          id_comercio: self.options[:id_comercio],
+          user: self.options[:user],
+          apikey: self.options[:apikey],
+          transaccion_id: options[:transaccion_id],
+          amount: amount(money)
+        }
+        post[:key_session] = self.options[:key_session]
+
+        post_to_json = post.to_json
+        puts(post_to_json)
+        post_to_json_encrypt = encrypt(post_to_json, self.options[:key_session])
+
+        final_post = post_to_json_encrypt + '-' + self.options[:user]
+        commit('capture', final_post)
+      end
+
+      def refund(money, authorization, options = {})
+        puts('---------------------------------------------')
+        puts('Refund')
+        puts('---------------------------------------------')
+        post = {
+          operation: 'Refund',
+          id_comercio: self.options[:id_comercio],
+          user: self.options[:user],
+          apikey: self.options[:apikey],
+          transaccion_id: options[:transaccion_id],
+          auth: authorization,
+          amount: amount(money)
+        }
+        post[:key_session] = self.options[:key_session]
+
+        post_to_json = post.to_json
+        puts(post_to_json)
+        post_to_json_encrypt = encrypt(post_to_json, self.options[:key_session])
+
+        final_post = post_to_json_encrypt + '-' + self.options[:user]
+        commit('refund', final_post)
+      end
+
+      # Currently unsupported
+      # def void(authorization, options={})
+      #   puts('---------------------------------------------')
+      #   puts('Void')
+      #   puts('---------------------------------------------')
+      #   refund(options["money"], authorization, options)
+      # end
+
+      def verify(credit_card, options = {})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def supports_scrubbing?
+        false
+      end
+
+      private
+
+      def add_customer_data(post, options)
+        post[:email] = options[:email] || 'nadie@mit.test'
+      end
+
+      def add_address(post, creditcard, options); end
```

Comment:
There is no reason to keep in methods that were in the template but not used in the adapter. `add_address` and `parse` can both be removed.